### PR TITLE
Remove mailing lists on user deletion

### PR DESCRIPTION
**Pull request by iszmais**
https://mantis.ilias.de/view.php?id=41275

- Delete all of users mailing lists when deleting a user.

### CODE REVIEW

File: Services/Contact/classes/class.ilMailingLists.php [hunk 1]
```diff
@@ -144,6 +144,25 @@ public function deleteTemporaryLists(): void
...
+    {
+        $res = $this->db->queryF(
+            'SELECT * FROM addressbook_mlist_ass as ass INNER JOIN addressbook_mlist as main ON main.ml_id = ass.ml_id WHERE usr_id = %s',
+            [ilDBConstants::T_INTEGER],
+            [$this->user->getId()]
+        );
+
+        $entries = [];
+        while ($row = $res->fetchRow(ilDBConstants::FETCHMODE_OBJECT)) {
+            $entries[] = new ilMailingList(new ilObjUser((int) $row->user_id), $row->ml_id);
```
Comment (mjansenDatabay) [inline Services/Contact/classes/class.ilMailingLists.php]:
```suggestion
            $entries[] = new ilMailingList(new ilObjUser((int) $row->user_id), (int) $row->ml_id);
```

File: Services/Contact/classes/class.ilMailingLists.php [hunk 2]
```diff
@@ -144,6 +144,25 @@ public function deleteTemporaryLists(): void
...
+    public function getAssigned(): array
+    {
+        $res = $this->db->queryF(
+            'SELECT * FROM addressbook_mlist_ass as ass INNER JOIN addressbook_mlist as main ON main.ml_id = ass.ml_id WHERE usr_id = %s',
+            [ilDBConstants::T_INTEGER],
+            [$this->user->getId()]
+        );
+
+        $entries = [];
+        while ($row = $res->fetchRow(ilDBConstants::FETCHMODE_OBJECT)) {
```
Comment (mjansenDatabay) [inline Services/Contact/classes/class.ilMailingLists.php]:
```suggestion
        while ($row = $ilDB->fetchObject($res)) {
```

File: Services/Contact/classes/class.ilMailingLists.php (hunk 1 above)
Comment (mjansenDatabay) [reply Services/Contact/classes/class.ilMailingLists.php]: Creating user instances here is very expensive in terms of performance. We should avoid this in scenarios where dozens of user accounts can potentially be deleted at once.

Please use simple SQL statements within `ilMailingLists`.

File: Services/Contact/classes/class.ilContactAppEventListener.php [hunk 3]
```diff
@@ -37,7 +37,19 @@ public static function handleEvent(string $a_component, string $a_event, array $
...
 
             $mailingLists = new ilMailingLists($user);
+            foreach ($mailingLists->getAll() as $mailingList) {
+                $mailingList->delete();
+            }
+            $assigned = $mailingLists->getAssigned();
             $mailingLists->deleteAssignments();
+            foreach ($assigned as $ass) {
+                if (ilBuddyList::getInstanceByUserId((int) $a_parameter['usr_id'])->getLinkedRelations()->isEmpty()) {
+                    $related = new ilMailingLists(new ilObjUser($ass->getUserId()));
```
Comment (mjansenDatabay) [inline Services/Contact/classes/class.ilContactAppEventListener.php]: Creating user instances is very expensive, we should avoid this in scenarios where dozens of user accounts can potentially be deleted at once.

Please use simple SQL statements within `ilMailingLists`.

**Review by mjansenDatabay — CHANGES_REQUESTED** [review]

**Comment by iszmais** [comment]
@mjansenDatabay I separated both problems in 2 different mantis issues.

Therefore this now only fix the missing deletion and all concerned scoped are deleted.
I will make a separate PR for the [other](https://mantis.ilias.de/view.php?id=41278)

File: Services/Contact/classes/class.ilContactAppEventListener.php [hunk 4]
```diff
@@ -37,6 +37,9 @@ public static function handleEvent(string $a_component, string $a_event, array $
             $user->setId((int) $a_parameter['usr_id']);
 
             $mailingLists = new ilMailingLists($user);
+            foreach ($mailingLists->getAll() as $mailingList) {
+                $mailingList->delete();
+            }
```
Comment (mjansenDatabay) [inline Services/Contact/classes/class.ilContactAppEventListener.php]: Please introduce a new method `deleteAll` which deletes all these lists with one single SQL statement.

**Review by mjansenDatabay — CHANGES_REQUESTED** [review]

**Review by mjansenDatabay — APPROVED** [review]

**Pull request merged**